release.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+set -e
+
+REPOSITORY="origin"
+BRANCH_FROM="main"
+while getopts ":a:f:b:r:k:" opt; do
+  case $opt in
+    a) RELEASE_API_VERSION="${OPTARG}"
+    ;;
+    f) RELEASE_VERSION="${OPTARG}"
+    ;;
+    b) BRANCH_FROM="${OPTARG}"
+    ;;
+    r) REPOSITORY="${OPTARG}"
+    ;;
+    k) GPG_KEY="${OPTARG}"
+    ;;
+
+    \?) echo "Invalid option -${OPTARG}" >&2
+    exit 1
+    ;;
+  esac
+
+  case ${OPTARG} in
+    -*) echo "Option $opt needs a valid argument"
+    exit 1
+    ;;
+  esac
+done
+
+if [[ -z "${GPG_KEY}" ]]; then
+    echo "GPG_KEY not set unable to sign the release. Please specify -k <YOUR_GPG_KEY>" 1>&2
+    exit 1
+fi
+
+if [[ -z ${RELEASE_API_VERSION} && -z ${RELEASE_VERSION} ]]; then
```

Comment:
should we make it an `||` with a comment until we sort out the api/framework interdependence?

Reply:
Both module release scripts validate their own versions are present so I'm tempted to leave it like this for now.